### ISSUE TITLE
aws_db_cluster_snapshot: Provider produced inconsistent result after apply 

### ISSUE THREAD

**Issue by vinced86** [terraform-providers/terraform-provider-aws#15096]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform AWS Provider Version
2.67 and 3.x

### Affected Resource(s)

aws_db_cluster_snapshot

### Terraform Configuration Files

```hcl
resource "aws_db_cluster_snapshot" "test" {
  db_cluster_identifier          =  "<arn of ressource>"
  db_cluster_snapshot_identifier = "Test"
}

```

### Expected Behavior

Snapshot created without error messages like this: 

Error: Provider produced inconsistent result after apply
When applying changes to aws_db_cluster_snapshot.externalbankuser_snapshot,
provider "registry.terraform.io/-/aws" produced an unexpected new value for
was present, but now absent.
This is a bug in the provider, which should be reported in the provider's own
issue tracker.
### Actual Behavior

I have this error message when i want to create aws_db_cluster_snapshot:

Error: Provider produced inconsistent result after apply
When applying changes to aws_db_cluster_snapshot.externalbankuser_snapshot,
provider "registry.terraform.io/-/aws" produced an unexpected new value for
was present, but now absent.
This is a bug in the provider, which should be reported in the provider's own
issue tracker.

### Steps to Reproduce

1. `terraform apply`

**Comment by angelabad** [comment]
Thank you @bflad  for the opportunity to learn

**Comment by bflad** [comment]
Additional validation for the `db_cluster_snapshot_identifier` argument to ensure its lowercase has been merged and will release with version 3.7.0 of the Terraform AWS Provider, later this week. Thanks to @angelabad for the implementation. 👍

**Comment by vinced86** [comment]
Thanks a lot for your helps!
